app/decorators/pseud_decorator.rb

```diff
@@ -0,0 +1,111 @@
+class PseudDecorator < SimpleDelegator
+
+  attr_reader :data
+
+  # TODO: pull this out into a reusable module
+  def self.decorate_from_search(results, search_hits)
+    search_data = search_hits.group_by { |doc| doc["_id"] }
+    results.map do |result|
+      data = search_data[result.id.to_s].first&.dig('_source') || {}
+      new_with_data(result, data)
+    end
+  end
+
+  def self.decorate(pseuds)
+    users = User.where(id: pseuds.map(&:user_id)).group_by(&:id)
+    work_counts
+    bookmark_counts
+    work_key = User.current_user.present? ? :general_works_count : :public_works_count
```

Comment:
I can never remember -- will `User.current_user.present?` be true if you're logged in as an admin? If not, should this be updated so admins will see the same work counts as users? (I know this isn't intended as an admin tool, but I have a feeling it will be useful at times.)

Reply:
Yep: https://github.com/otwcode/otwarchive/blob/master/app/controllers/application_controller.rb#L295